src/section/mod.rs

```diff
@@ -8,27 +8,395 @@
+        let secret_key_share = secret_key_set.secret_key_share(0);
+
+        // Note: `ElderInfo` is normally signed with the previous key, but as we are the first node
+        // of the network there is no previous key. Sign with the current key instead.
+        let elders_info = create_first_elders_info(&public_key_set, &secret_key_share, peer)?;
+
+        let mut section = Self::new(
+            SectionProofChain::new(elders_info.proof.public_key),
+            elders_info,
+        );
+
+        for peer in section.elders_info.value.peers() {
+            let member_info = MemberInfo::joined(*peer);
+            let proof = create_first_proof(&public_key_set, &secret_key_share, &member_info)?;
+            let _ = section.members.update(Proven {
+                value: member_info,
+                proof,
+            });
+        }
+
+        let section_key_share = SectionKeyShare {
+            public_key_set,
+            index: 0,
+            secret_key_share,
+        };
+
+        Ok((section, section_key_share))
+    }
+
+    pub fn merge(&mut self, other: Self) -> Result<()> {
+        if !other.chain.self_verify() {
+            return Err(Error::InvalidMessage);
+        }
+
+        self.chain
+            .merge(other.chain)
+            .map_err(|_| Error::UntrustedMessage)?;
+
+        match cmp_section_chain_position(&self.elders_info, &other.elders_info, &self.chain) {
+            Some(Ordering::Less) => {
+                self.elders_info = other.elders_info;
+            }
+            Some(Ordering::Equal)
+                if self.elders_info.value.elders.len() < other.elders_info.value.elders.len() =>
+            {
+                // Note: our `EldersInfo` is normally signed with the previous key, except the very
+                // first one which is signed with the latest key. This means that the first and
+                // second `EldersInfo`s are signed with the same key and so comparing only the keys
+                // is not enough to decide which one is newer. To break the ties, we use the fact
+                // that the first `EldersInfo` always has only one elder and the second one has
+                // two.
+                self.elders_info = other.elders_info;
+            }
+            Some(Ordering::Greater) | Some(Ordering::Equal) | None => (),
+        }
+
+        for info in other.members {
+            let _ = self.update_member(info);
+        }
+
+        self.members
+            .prune_not_matching(&self.elders_info.value.prefix);
+
+        Ok(())
+    }
+
+    /// Update the `EldersInfo` of our section.
+    pub fn update_elders(&mut self, new_elders_info: Proven<EldersInfo>) -> bool {
+        if !new_elders_info.verify(&self.chain) {
+            return false;
+        }
+
+        if new_elders_info != self.elders_info {
+            self.elders_info = new_elders_info;
+            self.members
+                .prune_not_matching(&self.elders_info.value.prefix);
```

Comment:
only `prune` ? no need to update?

Reply:
It's in the line above: `self.elders_info = new_elders_info`.

Reply:
but `self.elders_info = new_elders_info` only updates self.elders_info, not self.members ?

Reply:
Ah, I see. No, members are updated separately (it's the same reason as explained in the previous comment). We only do the prune here, because this is the place where our prefix can change.